src/TextArea/TextArea.less

```diff
@@ -19,34 +19,39 @@
   top: @size_xs;
   width: 100%;
   color: @neutral_dark_gray;
+}
 
-  .TextArea--label {
-    left: @size_s;
-    opacity: 1;
-  }
+.TextArea--label {
+  left: @size_s;
+  opacity: 1;
+  background-color: @neutral_white;
+  padding-right: @size_3xs;
```

Comment:
This right padding plus the removal of the left border on [line 47](https://github.com/Clever/components/pull/123/files#diff-f33c3b7a08d6659c4f1c4293d40f34cbR47) make the left alignments of text in the `TextArea` and `TextInput` components match.